static/gui/static/js/components/blockly.js

```diff
@@ -44,41 +44,33 @@ class PythonTextBox extends React.Component {
     this.download_python = this.download_python.bind(this);
     this.handleFileNameChange = this.handleFileNameChange.bind(this);
     this.handleFunctionNameChange = this.handleFunctionNameChange.bind(this);
-    this.handleScriptChange = this.handleScriptChange.bind(this);
-    this.handleTab = this.handleTab.bind(this);
     this.run_script = this.run_script.bind(this);
     this.upload = this.upload.bind(this);
     this.view_history = this.view_history.bind(this);
   }
-  /* Target function for the button "Cope Code". Set the text
-     in the editing box according to blockly. */
-  copy(event) {
-    document.getElementById("textarea").value = generatedPythonFromBlocklyBox.innerText;
-    this.setState({ pythonTextBoxCode: generatedPythonFromBlocklyBox.innerText });
+
+  /* Updates this class's state to contain the code specified in the parameter*/
+  updateCode(code)  {
+    this.setState({code});
   }
 
-  download_python(event) {
-    console.log("download listener");
-    event.preventDefault();
-    var element = document.createElement('a');
-    var filename = this.state.filename;
-    if (filename.substring(filename.length - 3) != ".py") {
-      filename += ".py";
-    }
-    element.setAttribute('href', 'data:text/plain;charset=utf-8,' + encodeURIComponent(this.state.pythonTextBoxCode));
-    element.setAttribute('download', filename);
-    element.style.display = 'none';
-    document.body.appendChild(element);
-    element.click();
-    document.body.removeChild(element);
+  /* Returns the CodeMirror editor object */
+  getEditor() {
+    return this.refs.editor.getCodeMirror();
+  }
+
+  /* Target function for the button "Copy Code". Set the text
+       in the editing box according to blockly. */
+  copy(event) {
+    this.getEditor().setValue(generatedPythonFromBlocklyBox.innerText);
   }
 
+  /* Function called whenver the custom function name is changed. */
   handleFunctionNameChange(event) {
     var _this = this;
-    var item = _this.props.customBlockList.find(element => element[0] === event.target.value); 
+    var item = _this.props.customBlockList.find(element => element[0] === event.target.value);
```

Comment:
Let's start using "let" instead of "var" everywhere?  I think that's the convention now because people prefer block scoping to function scoping.

Reply:
I agree. And maybe its also better to use 'const' for certain variables to indicate they won't be reassigned.